src/data/index.ts

```diff
@@ -220,6 +220,8 @@ export const exportToBackend = async (
       const urlString = url.toString();
 
       window.prompt(`🔒${t("alerts.uploadedSecurly")}`, urlString);
+    } else if (elements.length > appState.tooManyElements) {
+      window.alert(t("alerts.couldNotCreateShareableLinkTooBig"));
```

Comment:
This was annoying. It's good to see the message will be improved. Are we sure about tooManyElements for json server? Isn't it just json size, regardless of the number of elements?

Reply:
Yea, moreover we shouldn't make that decision on client. Instead, let's handle it on server and display an error message in case of upload failure.

Also, preview shouldn't ever be disabled just because you can't potentially upload to server.

Reply:
The canvas size error is a separate issue. There's an open PR for it https://github.com/excalidraw/excalidraw/pull/2210

Reply:
Ok my bad

Reply:
@Pierre-Monier Fortunately, it seems we're already responding with `RequestTooLargeError`, albeit in a weird `error_class` property instead of simply setting the status code to 413. Alas, better than nothing, and we can work with that.

![image](https://user-images.githubusercontent.com/5153846/95888902-ffe3e600-0d81-11eb-948b-34cfbf19eced.png)

All that remains is to catch that on client and display a friendly error message than what we're currently doing.

Reply:
Ok, I will do that and push it tonight